src/components/settings/SettingsLMSTab/LMSConfigs/Cornerstone/CornerstoneConfig.tsx

```diff
@@ -0,0 +1,199 @@
+import handleErrors from "../../../utils";
+import LmsApiService from "../../../../../data/services/LmsApiService";
+import { camelCaseDict, snakeCaseDict } from "../../../../../utils";
+import { CORNERSTONE_TYPE, SUBMIT_TOAST_MESSAGE } from "../../../data/constants";
+// @ts-ignore
+import ConfigActivatePage from "../ConfigBasePages/ConfigActivatePage.tsx";
+import CornerstoneConfigEnablePage, {
+  validations,
+  formFieldNames
+  // @ts-ignore
+} from "./CornerstoneConfigEnablePage.tsx";
+import type {
+  FormWorkflowButtonConfig,
+  FormWorkflowConfig,
+  FormWorkflowStep,
+  FormWorkflowHandlerArgs,
+} from "../../../../forms/FormWorkflow";
+import {
+  updateFormFieldsAction,
+  // @ts-ignore
+} from "../../../../forms/data/actions.ts";
+import type {
+  FormFieldValidation,
+} from "../../../../forms/FormContext";
+
+export type CornerstoneConfigCamelCase = {
+  displayName: string;
+  cornerstoneBaseUrl: string;
+  id: string;
+  active: boolean;
+  uuid: string;
+};
+
+export type CornerstoneConfigSnakeCase = {
+  display_name: string;
+  cornerstone_base_url: string;
+  id: string;
+  active: boolean;
+  uuid: string;
+  enterprise_customer: string;
+};
+
+export type CornerstoneFormConfigProps = {
+  enterpriseCustomerUuid: string;
+  existingData: CornerstoneConfigCamelCase;
+  existingConfigNames: string[];
+  onSubmit: (cornerstoneConfig: CornerstoneConfigCamelCase) => void;
+  onClickCancel: (submitted: boolean, status: string) => Promise<boolean>;
+};
+
+export const CornerstoneFormConfig = ({
+  enterpriseCustomerUuid,
+  onSubmit,
+  onClickCancel,
+  existingData,
+  existingConfigNames,
+}: CornerstoneFormConfigProps): FormWorkflowConfig<CornerstoneConfigCamelCase> => {
+  const configNames: string[] = existingConfigNames?.filter( (name) => name !== existingData.displayName);
+  const checkForDuplicateNames: FormFieldValidation = {
+    formFieldId: formFieldNames.DISPLAY_NAME,
+    validator: (formFields: CornerstoneConfigCamelCase) => {
+      return configNames?.includes(formFields.displayName)
+        ? "Display name already taken"
+        : false;
+    },
+  };
+
+  const saveChanges = async (
```

Comment:
We should be able to break this saveChanges function into a helper method, as the only differences are around the api methods called and the Types involved.

![image](https://user-images.githubusercontent.com/322346/229935525-c0145824-aa28-4035-8d9c-d2938941719d.png)